docs/source/Loaders.rst

```diff
@@ -80,3 +80,131 @@ test categories. With tags set, users can select a subset of the
+
+As the string after the ``:`` is a regular expression, three tests were
+filtered in. You can manipulate the regular expression to have only the
+test with that exact name::
+
+    $ avocado list /usr/share/avocado/tests/gdbtest.py:test_interactive$
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive
+
+The regular expression enables you to have more complex filters.
+Example::
+
+    $ avocado list /usr/share/avocado/tests/gdbtest.py:GdbTest.test_[le].*raw
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_existing_commands_raw
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_load_set_breakpoint_run_exit_raw
+
+Once the test reference is providing you the expected outcome, you can
+replace the ``list`` subcommand with the ``run`` subcommand to execute your
+tests::
+
+    $ avocado run /usr/share/avocado/tests/gdbtest.py:GdbTest.test_[le].*raw
+    JOB ID     : 333912fb02698ed5339a400b832795a80757b8af
+    JOB LOG    : $HOME/avocado/job-results/job-2017-06-14T14.54-333912f/job.log
+     (1/2) /usr/share/avocado/tests/gdbtest.py:GdbTest.test_existing_commands_raw: PASS (0.59 s)
+     (2/2) /usr/share/avocado/tests/gdbtest.py:GdbTest.test_load_set_breakpoint_run_exit_raw: PASS (0.42 s)
+    RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
+    JOB TIME   : 1.15 s
+    JOB HTML   : $HOME/avocado/job-results/job-2017-06-14T14.54-333912f/html/results.html
+
+.. warning:: Specially when using regular expressions, it's recommended
+   to individually enclose your Test References in quotes to avoid bash
+   of corrupting them. In that case, the command from the example above
+   would be:
+   ``avocado run "/usr/share/avocado/tests/gdbtest.py:GdbTest.test_[le].*raw"``
+
+External Loader
+---------------
+
+Using the External Loader, Avocado will consider that and External
+Runner will be in place and so Avocado doesn't really need to resolve
+the references. Instead, Avocado will pass the references as parameters
+to the External Runner. Example::
+
+    $ avocado run 20
+    Unable to resolve reference(s) '20' with plugins(s) 'file', 'robot',
+    'vt', 'external', try running 'avocado list -V 20' to see the details.
+
+In the command above, no loaders can resolve ``20`` as a test. But running
+the command above with the External Runner ``/bin/sleep`` will make Avocado
+to actually execute ``/bin/sleep 20`` and check for its return code::
+
+    $ avocado run 20 --loaders external:/bin/sleep
+    JOB ID     : 42215ece2894134fb9379ee564aa00f1d1d6cb91
+    JOB LOG    : $HOME/avocado/job-results/job-2017-06-19T11.17-42215ec/job.log
+     (1/1) 20: PASS (20.03 s)
+    RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
+    JOB TIME   : 20.13 s
+    JOB HTML   : $HOME/avocado/job-results/job-2017-06-19T11.17-42215ec/html/results.html
+
+.. warning:: It's safer to put your Test References at the end of the
```

Comment:
I believe we already have such warning elsewhere, I meant only to use it (I actually wanted to send a patch to do that everywhere, people learn by examples ...). Anyway I'm fine with this as well...